api/docs/bt.dox

```diff
@@ -89,6 +89,19 @@ a primary contributor to DynamoRIO's efficiency.
 The instruction representation includes all of the operands, whether
 implicit or explicit, and the condition code effects of each instruction.
 This allows for analysis of liveness of registers and condition codes.
+The operands are split into sources and destinations.
+
+A memory reference is treated as one operand even when it uses
+registers to compute its address: those constituent registers are not
+listed as their own separate source operands (unless they are read for
+other reasons such as updating the index register).  This means that a
+store to memory will have that store as a destination operand without
+listing the store's addressing mode registers as source operands in
+their own right.  Tools interested in all registers inside such
+operands can use interfaces like instr_reads_from_reg() to query (it
+will consider registers used to compute a store address as being read)
+or opnd_get_reg_used() to generically walk the registers inside an
```

Comment:
We might want to mention `opnd_get_num_regs_used()` as well, since `opnd_get_reg_used()` also needs an index.
Although, "Used in conjunction with opnd_num_regs_used()" is already mentioned in doxygen for `opnd_get_reg_used()`.

Reply:
Your earlier comments seem to have disappeared: you had one about include at least a ref to #opnd_t which is indeed missing.  I added it to the 1st paragraph which introduces #instr_t.

I reworded to move the instr_reads_from_reg() to be mentioned last to avoid your point about it not being the same as querying/iterating all the registers.

I added opnd_get_num_reg_used() as requested.

Reply:
Oh, yes I removed the whole discussion about `opnd_get_num_reg_used()` and `opnd_get_reg_used()` because I realized you talked about it right after `instr_reads_from_reg()`.
The reordering (talking about those two #opnd_t related functions before `instr_reads_from_reg()`) flows better with the whole explanation of "querying registers from an operand".
Mentioning #opnd_t I didn't think it was critical, but now that it's in, it reads better.